src/android/plugin/google/maps/PluginMarker.java

```diff
@@ -710,6 +715,32 @@ private void removeMultipleMarkers(final JSONArray args, final CallbackContext c
     callbackContext.success();
   }
 
+  @SuppressWarnings("unused")
+  private void removeMultipleMarkersByType(final JSONArray args, final CallbackContext callbackContext) throws JSONException {
+    String typeToRemove = args.getString(1);
+    HashMap<String, Object> objectsCopy = new HashMap<String, Object>(this.objects);
+    JSONArray markersToRemove = new JSONArray();
+
+    for (HashMap.Entry<String, Object> entry : objectsCopy.entrySet()) {
+      Object properties = entry.getValue();
+      if (properties instanceof JSONObject) {
+        if (((JSONObject)properties).has("type")) {
```

Comment:
Maybe we could've add a variable like ` JSONObject jsonObject =  (JSONObject)properties ` so you can save some of those ugly castings.